main.go

```diff
@@ -104,15 +115,71 @@ func initLogging() (*os.File, error) {
 	return file, nil
 }
 
+// extension specific PublicSettings
+type PublicSettings struct {
+	Script   string   `json:"script"`
+	FileURLs []string `json:"fileUris"`
+}
+
+// extension specific ProtectedSettings
+type ProtectedSettings struct {
+	SecretString       string   `json:"secretString"`
+	SecretScript       string   `json:"secretScript"`
+	FileURLs           []string `json:"fileUris"`
+	StorageAccountName string   `json:"storageAccountName"`
+	StorageAccountKey  string   `json:"storageAccountKey"`
+}
+
 func main() {
-	// TODO Add more robust error handling
 	file, err := initLogging()
 	if err != nil {
 		fmt.Println("Error opening the provided logfile.")
-		os.Exit(1)
+		os.Exit(-1)
 	}
 	defer file.Close()
 
+	extensionMrseq, environmentMrseq, err := sequence.GetMostRecentSequenceNumber()
+	if err != nil {
+		errorLogger.Printf("%+v", err)
+		os.Exit(-1)
```

Comment:
Make -1 a constant with a verbose name and use that everywhere for exit.
Eg: 
```
EXIT_ERROR = -1
```

Reply:
Another idea would be to have distinct exit codes for distinct failures, this helps in debugging on what went wrong just based on the exit code.
Eg: 
```
UNKNOWN_ERROR = -1
MRSEQ_NOT_FOUND = -2
SHOULD_NOT_RUN = -3
```

And so on and so forth